src/phantom/_utils/types.py

```diff
@@ -1,6 +1,10 @@
+import sys
 from typing import TypeVar
 
-from numerary.types import CachingProtocolMeta  # type: ignore[attr-defined]
+if sys.version_info > (3, 7):
+    from numerary.protocol import CachingProtocolMeta
+else:
+    from numerary.types import CachingProtocolMeta  # type: ignore[attr-defined]
```

Comment:
Gonna have to double check that this would in fact fix the caching issue

Reply:
If it doesn't, perhaps worth trying with `if sys.version_info > (3, 7) or TYPE_CHECKING:`?

Reply:
That's an interesting idea!

Reply:
Oh, seems it did not in fact work: https://github.com/antonagestam/phantom-types/actions/runs/3483387875/jobs/5826831019

Reply:
> If it doesn't, perhaps worth trying with `if sys.version_info > (3, 7) or TYPE_CHECKING:`?

Hm, but I think by doing this, it'd eliminate the possibility of type checking successfully with < 3.8 ?

Reply:
Hmm, I thought the caching issue only occurs with numerary>=0.4.4? I'll gladly admit I haven't fully understood yet what's going on there though 🫣

Reply:
I can't say I've bisected or so to figure out if it works properly with older versions or what triggers what more exactly.

It's a bit difficult due to cached mypy runs etc.

At least it seems the ignored error is cause, due to what type mypy decides to write to cache(`Any`) and that with latest pypi version of `numerary`

Reply:
Gotcha, I think this fix is good anyways so we can move forward it. I do need to fix sphinx before I can make a release though ...